File: backend/backend/models/lesson/lesson.js
const mongoose = require("mongoose");

const lessonSchema = mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
    },
    isCompleted: {
      type: Boolean,
      default: false,
    },
    isUnlocked: {
      type: Boolean,
      default: false,
    },
    questionId: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Question",
      },
    ],
    sectionId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Section",
    },
  },
  { timestamp: true }
);

// export
const Lesson = mongoose.model("Lesson", lessonSchema);
module.exports = Lesson;

File: frontend/doulingo/lib/domain/section/entities/lesson.dart
class LessonEntity {
  final String? id;
  final String? title;
  final bool? isCompleted;
  final bool? isUnlocked;
  final List<dynamic>? questions;

  LessonEntity({
    this.id,
    this.title,
    this.isCompleted,
    this.isUnlocked,
    this.questions,
  });
}
